### ISSUE TITLE
DropdownMenu component accessibility issues

### ISSUE THREAD

**Issue by jbpenrath** [suitenumerique/ui-kit#183]
### Bug Report

**Problematic behavior**

DropdownMenu (and surely ContextMenu) has accessibilty issues. It uses `role="dialog"`, is a focus trap, and has only arrow-key navigation (no tab). They are neither properly announced nor operable via keyboard.
Furthermore, action item are not interactive component (button) but div.
Then the trigger component is missing the aria attribute `aria-expanded` to inform about its open state.
Selected items are not announced has selected (e.g use `<input type="radio">` in `<fieldset>/<legend>`).
Dropdown items receive a gray background on focus with a contrast ratio of only **1.1:1** (`#F3F3F4` on `#FFFFFF`)

**Expected behavior/code**
All interactive components must be operable via keyboard and correctly announced by screen readers, following [WAI-ARIA Authoring Practices](https://www.w3.org/WAI/ARIA/apg/).

**Steps to Reproduce**
1. Use Tab to navigate within a DropdownMenu
2. Press `[Enter]` to open — dropdown opens but focus management is broken
3. Press `[Tab]` — Nothing happen and focus is trapped within the DropdownMenu
4. Press `[Escape]` — may or may not close
5. Repeat with screen reader — roles/states are not correctly announced

**Environment**
- RGAA Criterion: 7.1, 7.3, 10.7, 12.9
- Severity: 🔴 Blocking

**Additional context/Screenshots**
RGAA reference: 
  - [Criterion 7.1](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#7.1)
  - [Criterion 7.3](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#7.3)
  - [Criterion 10.7](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#10.7)
  - [Criterion 12.9](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#12.9)
